src/Core/maps/src/Platform/iOS/MauiMKMapView.cs

```diff
@@ -0,0 +1,305 @@
+﻿using System;
+using System.Collections;
+using System.Linq;
+using CoreLocation;
+using MapKit;
+using Microsoft.Maui.Maps.Handlers;
+using Microsoft.Maui.Platform;
+using ObjCRuntime;
+using UIKit;
+
+namespace Microsoft.Maui.Maps.Platform
+{
+	internal class MauiMKMapView : MKMapView
+	{
+		IMapHandler? _handler;
+		object? _lastTouchedView;
+
+		public MauiMKMapView(IMapHandler handler)
+		{
+			_handler = handler;
+			OverlayRenderer = GetViewForOverlay1;
+		}
+
+		protected override void Dispose(bool disposing)
+		{
+			_handler = null;
+			base.Dispose(disposing);
+		}
+
+		public override void MovedToWindow()
+		{
+			base.MovedToWindow();
+			if (Window != null)
+				Startup();
+			else
+				Cleanup();
+		}
+
+		void Startup()
+		{
+			RegionChanged += MkMapViewOnRegionChanged;
+			DidSelectAnnotationView += MkMapViewOnAnnotationViewSelected;
+		}
+
+		void Cleanup()
+		{
+			if (_handler == null)
+				return;
+
+			RegionChanged -= MkMapViewOnRegionChanged;
+			DidSelectAnnotationView -= MkMapViewOnAnnotationViewSelected;
+			_handler = null;
```

Comment:
I wouldn't set the `_handler` to null here though

The map might get removed and then added back to the window for various reasons. 

For example in shell if the user is switching tabs or flyouts

Reply:
Done